### PR TITLE
fix: make importing workspace-specific objects as copy

### PR DESCRIPTION
**Pull request by SuZhou-Joe**
### Description

1. Restore skipped collapsible_nav test cases.
2. Optimization when workspace_id is not valid.
3. Fix import logic when importing.
4. Fix reference id issue when importing to target workspaces.

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/core/server/saved_objects/service/lib/repository.ts [hunk 1]
```diff
@@ -606,17 +657,24 @@ export class SavedObjectsRepository {
...
-              metadata: { isNotOverwritable: true },
-            }),
-          },
-        });
+        const transformedObject = this._serializer.rawToSavedObject(doc as SavedObjectsRawDoc);
+        const filteredWorkspaces = SavedObjectsUtils.filterWorkspacesAccordingToBaseWorkspaces(
+          options.workspaces,
+          transformedObject.workspaces
+        );
+        if (!filteredWorkspaces.length) {
```
Comment (wanglam) [inline src/core/server/saved_objects/service/lib/repository.ts:665]: Is that means the doc will be treated conflict if doc.workspaces contains all options.workspaces?

File: src/core/server/saved_objects/import/regenerate_ids.ts [hunk 2]
```diff
@@ -42,3 +44,36 @@ export const regenerateIds = (objects: SavedObject[]) => {
...
+  objectLimit: number;
+}): Promise<Map<string, { id?: string; omitOriginId?: boolean }>> => {
+  const { savedObjects, savedObjectsClient, workspaces, objectLimit } = props;
+  if (!workspaces || !workspaces.length) {
+    return savedObjects.reduce((acc, object) => {
+      return acc.set(`${object.type}:${object.id}`, { id: object.id, omitOriginId: false });
+    }, new Map<string, { id: string; omitOriginId?: boolean }>());
+  }
+
+  const allSavedObjects = (await exportSavedObjects({
```
Comment (wanglam) [inline src/core/server/saved_objects/import/regenerate_ids.ts:61]: Why do we need exportSavedObjects again here?

File: src/plugins/workspace/public/workspace_client.ts [hunk 3]
```diff
@@ -160,6 +162,8 @@ export class WorkspaceClient {
     if (result?.success) {
       this.workspaces.workspaceList$.next(result.result.workspaces);
     }
+
+    this.workspaces.hasFetchedWorkspaceList$.next(true);
```
[the remark this answers is not in the document]
Comment (SuZhou-Joe) [reply src/plugins/workspace/public/workspace_client.ts:166]: That works as well but I and Yulong thought that creating a explicit variable to indicate the initial fetch makes more sense. And changing the type of `workspaceList` may lead to many side effect on consumer side.

File: src/core/server/saved_objects/service/lib/repository.ts (hunk 1 above)
Comment (SuZhou-Joe) [reply src/core/server/saved_objects/service/lib/repository.ts:665]: Yes.

File: src/core/server/saved_objects/import/regenerate_ids.ts (hunk 2 above)
Comment (SuZhou-Joe) [reply src/core/server/saved_objects/import/regenerate_ids.ts:61]: You are right, there is no need to call `exportSavedObjects` here, thanks for catching.

File: src/plugins/workspace/public/components/workspace_fatal_error/workspace_fatal_error.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import React from 'react';
+import { FormattedMessage } from '@osd/i18n/react';
+import { useOpenSearchDashboards } from '../../../../opensearch_dashboards_react/public';
+
+export function WorkspaceFatalError(props: { error?: string }) {
+  const {
+    services: { workspaces, application },
+  } = useOpenSearchDashboards();
+  const goBackToHome = () => {
+    window.location.href =
```
[the remark this answers is not in the document]
Comment (SuZhou-Joe) [reply src/plugins/workspace/public/components/workspace_fatal_error/workspace_fatal_error.tsx]: Fixed.

File: src/plugins/workspace/public/workspace_client.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (wanglam) [reply src/plugins/workspace/public/workspace_client.ts:166]: Do we need to update `hasFetchedWorkspaceList` equal true when result succeed?

File: src/core/server/saved_objects/import/regenerate_ids.ts [hunk 5]
```diff
@@ -42,3 +43,39 @@ export const regenerateIds = (objects: SavedObject[]) => {
...
+  objectLimit: number;
+}): Promise<Map<string, { id?: string; omitOriginId?: boolean }>> => {
+  const { savedObjects, savedObjectsClient, workspaces } = props;
+  if (!workspaces || !workspaces.length) {
+    return savedObjects.reduce((acc, object) => {
+      return acc.set(`${object.type}:${object.id}`, { id: object.id, omitOriginId: false });
+    }, new Map<string, { id: string; omitOriginId?: boolean }>());
+  }
+
+  const bulkGetResult = await savedObjectsClient.bulkGet(
```
Comment (wanglam) [inline src/core/server/saved_objects/import/regenerate_ids.ts:60]: Could we just use the passed `savedObjects` here?

Comment (SuZhou-Joe) [reply src/core/server/saved_objects/import/regenerate_ids.ts:60]: We can not, the `savedObjects` here is what in user's import file, the workspaces info may not be the same as what is in kibana index.

File: src/plugins/workspace/public/workspace_client.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (SuZhou-Joe) [reply src/plugins/workspace/public/workspace_client.ts:166]: We should. It indicate if we fetch the list, no matter if it is success or not.

**Review by wanglam — APPROVED** [review]

**Pull request merged**